GameServerLib/Scripting/CSharp/CSharpScriptEngine.cs

```diff
@@ -53,50 +72,58 @@ public bool Load(List<string> scriptLocations)
             var assemblyName = Path.GetRandomFileName();
 
             var references = new List<MetadataReference>();
-            foreach (var a in AppDomain.CurrentDomain.GetAssemblies())
-                if (!a.IsDynamic && !a.Location.Equals(""))
-                    references.Add(MetadataReference.CreateFromFile(a.Location));
+            foreach (var assembly in AppDomain.CurrentDomain.GetAssemblies())
+            {
+                if (!assembly.IsDynamic && !assembly.Location.Equals(""))
+                {
+                    references.Add(MetadataReference.CreateFromFile(assembly.Location));
+                }
+            }
+            //Add vector2 reference because vector2 isn't in System.Numerics in dot net core
```

Comment:
This isn't true for either [.NET Core](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.vector2?view=netcore-3.1), nor [.NET Framework ](https://docs.microsoft.com/en-us/dotnet/api/system.numerics.vector2?view=netframework-4.8).

Reply:
It unfortunately is true for .net core on mac. At least on our current version of .net core which is 3.0.
https://github.com/dotnet/standard/issues/783